src/workspace/src/logger_text/src/logger_text.cpp

```diff
@@ -5,14 +5,37 @@
 #include <cstdio>
 #include <numeric>
 
-Writer::Writer(std::string & outputFolder, std::string separator):outputFolder(outputFolder),separator(separator),transformListener(buffer){
 
+Writer::Writer(std::string & outputFolder, std::string separator):outputFolder(outputFolder),separator(separator),transformListener(buffer){
+			configurationClient = node.serviceClient<setting_msg::ConfigurationService>("get_configuration");
+			ROS_INFO("getting speed threshold configuration...");
+			getSpeedThresholdConfig();
 }
 
 Writer::~Writer(){
 	finalize();
 }
 
+void Writer::getSpeedThresholdConfig(){
+	setting_msg::ConfigurationService srv;
+
+    srv.request.key = "speedThresholdKmh";
+
+    if(configurationClient.call(srv)){
+    	std::string speedThresKmh = srv.response.value;
+    	ROS_INFO_STREAM("speed threshold from config file : "<< speedThresKmh);
+        speedThresholdKmh = stod(speedThresKmh);
```

Comment:
stod will throw an exception with an invalid string, crashing the whole thing. Add error checking and/or proper exception handling.

Reply:
fix